users/raissi/setups/common/decoder/BASE_factored_hybrid_search.py

```diff
@@ -917,13 +919,33 @@ def recognize(
             is_count_based=True,
         )
 
+        adv_search_extra_config = (
+            copy.deepcopy(adv_search_extra_config)
+            if adv_search_extra_config is not None
+            else rasr.RasrConfig()
+        )
         if search_parameters.altas is not None:
-            adv_search_extra_config = rasr.RasrConfig()
             adv_search_extra_config.flf_lattice_tool.network.recognizer.recognizer.acoustic_lookahead_temporal_approximation_scale = (
                 search_parameters.altas
             )
-        else:
-            adv_search_extra_config = None
+        if search_parameters.lm_lookahead_scale is not None:
+            # Use 4gram for lookahead. The lookahead LM must not be too good.
+            #
+            # Half the normal LM scale is a good starting value.
+
+            # To validate the assumption the original LM is a 4gram
+            assert original_lm_config.language_model_config.type.lower() == "arpa"
```

Comment:
@NeoLegends JFYI, here you should do: ` assert original_lm_config.type.lower() == "arpa"` instead